xc/device.ml

```diff
@@ -608,11 +610,32 @@ let qemu_media_change ~xs device _type params =
 	if is_upstream_qemu device.frontend.domid
 	then begin
 		let cd = "ide1-cd1" in
-		let qmp_cmd =
-			if params = ""
-			then (Qmp.Command(None, Qmp.Eject (cd, Some true)))
-			else (Qmp.Command(None, Qmp.Change (cd, params, None))) in
-		qmp_write device.frontend.domid qmp_cmd
+		match params with
```

Comment:
Thanks @mg12 
I have put `qmp_read` inside of `qmp_write_and_read`, please see:
https://github.com/xapi-project/xenopsd/pull/367/files#diff-7a8f9b276e7662a150bdc8e87d9de515R312

> if this doesn't block indefinitely waiting for the next message sent by qemu, it will not detect immediately that a new qmp event is available when qemu sends a message.
This doesn't block indefinitely. Yes, it will not detect some events immediately.